sky/serve/core.py

```diff
@@ -160,17 +174,21 @@ def up(
         controller_exist = (
             global_user_state.get_cluster_from_name(controller_name)
             is not None)
-        requested_cloud = list(task.resources)[0].cloud
-        controller_cloud = (requested_cloud if not controller_exist and
-                            controller_resources.cloud is None else
-                            controller_resources.cloud)
+        if (not controller_exist and
+                controller_resources_in_config.cloud is None):
+            controller_clouds = requested_clouds
+        else:
+            controller_clouds = {controller_resources_in_config.cloud}
```

Comment:
If `controller_exist` and `controller_resources_in_config.cloud is None`, the `controller_clouds` will still be set to None?

Reply:
Yeah, because if the controller exists, we don't want to trigger a resources not match error for the controller. For example, the controller exists and it is on AWS, and no controller resources in config is found; while now I try to `sky serve up` a service w/ replica on GCP. In such case, we want to directly use the controller, instead of setting controller cloud to GCP and causing a resources not match on controller.